pkg/controller/podautoscaler/horizontal.go

```diff
@@ -159,11 +159,16 @@ func (a *HorizontalController) computeReplicasForCPUUtilization(hpa *autoscaling
 	utilization := int32(*currentUtilization)
 
 	usageRatio := float64(utilization) / float64(targetUtilization)
-	if math.Abs(1.0-usageRatio) > tolerance {
-		return int32(math.Ceil(usageRatio * float64(currentReplicas))), &utilization, timestamp, nil
+	if math.Abs(1.0-usageRatio) <= tolerance {
+		return currentReplicas, &utilization, timestamp, nil
 	}
 
-	return currentReplicas, &utilization, timestamp, nil
+	desiredReplicas := math.Ceil(usageRatio * float64(numRunningPods))
```

Comment:
this could effectively cause a scale down, even if it looks like a scale up (if `desiredReplicas * numRunningPods < (numRunningPods + numPendingPods)`)?  Should we deal with that, or at least log or note it somewhere?